backend/services/evidence_metadata.go

```diff
@@ -43,6 +43,10 @@ func CreateEvidenceMetadata(ctx context.Context, db *database.Connection, i Edit
 		return backend.WrapError("Unwilling to create evidence metadata", backend.UnauthorizedWriteErr(err))
 	}
 
+	if i.Source == "" || i.Body == "" {
```

Comment:
It's up to you if this is the way you want to do that, but generally you'd want to do these checks individually, as it'll provide more of an indication to the caller how to fix the issue. 

You also forgot to add a similar check in updateEvidence. I'd need to check, but I suspect you could still create empty evidence if you wanted by creating evidence with source and body, and then editing the evidence to remove the source and body.

Reply:
You might also look at `strings.TrimSpace`